server/src/main/java/org/elasticsearch/index/mapper/BlockSourceReader.java

```diff
@@ -149,6 +192,75 @@ public String toString() {
         }
     }
 
+    private static class Points extends BlockSourceReader {
+
+        Points(ValueFetcher fetcher) {
+            super(fetcher);
+        }
+
+        @Override
+        protected void append(BlockLoader.Builder builder, Object v) {
+            if (v instanceof SpatialPoint point) {
+                ((BlockLoader.PointBuilder) builder).appendPoint(point);
+            } else if (v instanceof String wkt) {
+                try {
+                    // TODO: figure out why this is not already happening in the GeoPointFieldMapper
+                    Geometry geometry = WellKnownText.fromWKT(GeometryValidator.NOOP, false, wkt);
+                    if (geometry instanceof Point point) {
+                        // TODO: perhaps we should not create points for later GC here, and pass in primitives only?
+                        ((BlockLoader.PointBuilder) builder).appendPoint(new SpatialPoint(point.getX(), point.getY()));
+                    } else {
+                        throw new IllegalArgumentException("Cannot convert geometry into point:: " + geometry.type());
+                    }
+                } catch (Exception e) {
```

Comment:
Probably best to catch precise exceptions here.

Reply:
Done. It was two different exceptions, so I'll use `catch (IOException | ParseException e)`